src/main/java/com/zendesk/maxwell/producer/MaxwellRedisProducer.java

```diff
@@ -6,42 +6,98 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import redis.clients.jedis.Jedis;
+import redis.clients.jedis.StreamEntryID;
 import redis.clients.jedis.exceptions.JedisConnectionException;
+import java.util.HashMap;
+import java.util.Map;
 
 public class MaxwellRedisProducer extends AbstractProducer implements StoppableTask {
 	private static final Logger logger = LoggerFactory.getLogger(MaxwellRedisProducer.class);
 	private final String channel;
-	private final String listkey;
-	private final String redistype;
+	private final String redisType;
 	private final Jedis jedis;
 
-	public MaxwellRedisProducer(MaxwellContext context, String redisPubChannel, String redisListKey, String redisType) {
+	@Deprecated
+    public MaxwellRedisProducer(MaxwellContext context, String redisPubChannel, String redisListKey, String redisType) {
+		this(context, redisType);
+	}
+
+	public MaxwellRedisProducer(MaxwellContext context, String redisType) {
 		super(context);
 
-		channel = redisPubChannel;
-		listkey = redisListKey;
-		redistype = redisType;
+		if (this.context.getConfig().redisListKey != null) {
+			channel = context.getConfig().redisListKey;
+		}
+		else if (this.context.getConfig().redisStreamKey != null) {
+			channel = context.getConfig().redisStreamKey;
+		}
+		else {
+			channel = this.context.getConfig().redisPubChannel;
+		}
+
+		this.redisType = redisType;
 
 		jedis = new Jedis(context.getConfig().redisHost, context.getConfig().redisPort);
 		jedis.connect();
+
 		if (context.getConfig().redisAuth != null) {
 			jedis.auth(context.getConfig().redisAuth);
 		}
+
 		if (context.getConfig().redisDatabase > 0) {
 			jedis.select(context.getConfig().redisDatabase);
 		}
 	}
 
-	private void sendToRedis(String msg) {
-		switch (redistype) {
+	private void sendToRedis(RowMap msg) throws Exception {
+		String messageStr = msg.toJSON(outputConfig);
+
+		switch (redisType) {
 			case "lpush":
-				jedis.lpush(this.listkey, msg);
+				jedis.lpush(this.channel, messageStr);
+				break;
+			case "xadd":
+				Map<String, String> message = new HashMap<>();
+
+				String jsonKey = this.context.getConfig().redisStreamJsonKey;
+				
+				if (jsonKey == null) {
+					// TODO dot notated map impl in RowMap.toJson
+					throw new IllegalArgumentException("Stream requires key name for serialized JSON value");
+				}
+				else {
+					message.put(jsonKey, messageStr);
+				}
+
+				// TODO timestamp resolution coercion
+				// 			Seconds or milliseconds, never mixing precision
+				//      	DML events will natively emit millisecond precision timestamps
+				//      	CDC events will natively emit second precision timestamp
+				// TODO configuration option for if we want the msg timestamp to become the message ID
+				//			Requires completion of previous TODO
+				jedis.xadd(this.channel, new StreamEntryID(), message);
```

Comment:
it's strange to me that Jedis forces you to generate your own `StreamEntryID` -- according to https://redis.io/commands/xadd the vastly preferred method of ID-ing these things is to just pass "*" to redis and let it generate its own id.  In particular, I'd worry about:

> When a user specified an explicit ID to XADD,  ... the user must specify an ID which is greater than any other ID currently inside the stream, otherwise the command will fail. 

which means that two producers sending to the same stream can and will screw each other up. 

...

Ah.  Ok.  further source code reading indicates that you should be passing `StreamEntryID.NEW_ENTRY` instead of what you're doing.

Reply:
Thanks for pointing that out. Would have produced:
ERR The ID specified in XADD is equal or smaller than the target stream top item